### DISCUSSION TITLE
Inject probe laser at an angle in LWFA

### DISCUSSION THREAD

**Discussion by ycaophysics** [BLAST-WarpX/warpx#4953] [Simulation Inputs]
Hi. I've been trying to add a second laser in 2D/3D LWFA simulation at an angle from 60 to 90 degrees. 
The laser needs to collide with the first trailing wakefield bubble. 
After multiple trials I found that: 
1. the laser antenna has to be inside the initial starting window or it doesn't show up. 
2. with moving window the probe laser will experience relativistic effects, when its longitudinal speed is 0. 
3. It doesn't seem to accept a huge frame that is not moving for LWFA. The error shows: 
![image](https://github.com/ECP-WarpX/WarpX/assets/70364621/bd274cbd-8164-44b6-b9f5-39e6b0f33459)
And I don't necessarily want to make the transverse laser move artificially with the moving frame.

Here is my code using PICMI. Thank you very much!
[ut3_warpx.txt](https://github.com/ECP-WarpX/WarpX/files/15458317/ut3_warpx.txt)

Also by the way, how does the blocking factor affect the simulation (both in its computing accuracy and the computing speed? (At the same simulation box)

**Comment by ax3l** [comment]
> Also by the way, how does the blocking factor affect the simulation (both in its computing accuracy and the computing speed? (At the same simulation box)


For your last question, does this help?
https://warpx.readthedocs.io/en/latest/usage/workflows/domain_decomposition.html

**Reply by ycaophysics** [reply]
Hi Aexl:
 So the domain decomposition doesn't actually affect the physics right?
 
By the way, there's a problem I saw right now. I injected an additional pulse at around 60 degree angle. However, when I do the diagnosis in a non-moving frame, it shows like this picture: 
<img width="839" alt="屏幕截图 2024-06-03 180820" src="https://github.com/ECP-WarpX/WarpX/assets/70364621/08fcf0c1-6791-40f2-9ee6-469f418555c2">
The transverse beam doesn't seem to be injected correctly with antenna. How would you think I should fix it?
Here's the input file:
[inputs_BNL_stable.txt](https://github.com/user-attachments/files/15540701/inputs_BNL_stable.txt)

Thanks
Yuxuan Cao
